assembly/src/parsers/field_ops.rs

```diff
@@ -164,6 +164,31 @@ pub(super) fn parse_pow2(
     Ok(())
 }
 
+/// Translates exp assembly instructions to VM operations.
+///
+/// Appends a sequence of operations to raise the base(second element) to the power specified by
+/// the top element in the stack.
+///
+/// VM cycles:
+/// - 71 cycles
```

Comment:
Not for this PR: in the future we should allow passing parameters to this instruction. There are a couple of options I have for this, but not sure what's the best way to do it is.

* Option 1: provide the number of expected bits for the exponent - e.g., `exp.32` would perform `EXPACC` only 32 times.
* Option 2: provide the exponent  - e.g., `exp.1234` would compute `a^1234` where `a` is the value on the top of the stack.

It would be great to somehow be able to specify both - e.g., `exp.32b` means exponent on the stack is 32 bits but `exp.32` means compute `a^32` - but I'm not sure what's the best way to do this yet.

Reply:
Both option makes sense! I will take it up post this PR.

Reply:
Thinking about the naming convention a bit more, I think it could be like so:

* We could use the `u` prefix to indicate the number of bits. For example: `exp.u32` would indicate that the exponent is expected to contain a 32-bit value, `exp.u8` would indicate that it is expected to contain an 8-bit value etc.
* Using a simple value would indicate that the exponent is provided via the immediate value: `exp.32` would mean computing a^32, `exp.8` would mean computing `a^8` etc.